src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3016,8 +3016,16 @@ void CodeGen::genCodeForCpBlkUnroll(GenTreeBlk* node)
         regNumber tempReg = node->GetSingleTempReg(RBM_ALLFLOAT);
 
         instruction simdMov = simdUnalignedMovIns();
-        for (unsigned regSize = XMM_REGSIZE_BYTES; size >= regSize;
-             size -= regSize, srcOffset += regSize, dstOffset += regSize)
+
+        // Get the largest SIMD register available if the size is large enough
+        unsigned regSize = size >= YMM_REGSIZE_BYTES ? compiler->getSIMDVectorRegisterByteLength() : XMM_REGSIZE_BYTES;
```

Comment:
We could do this on machines that only have AVX, but not AVX2 (few as they are) by doing `compOpportunisticallyDependsOn(InstructionSet_AVX)` instead.

Reply:
Do you mean check `compOpportunisticallyDependsOn(InstructionSet_AVX)` instead of checking the `size >= YMM_REGSIZE_BYTES` condition?

Reply:
I meant something like
```cpp
unsigned regSize = (size >= YMM_REGSIZE_BYTES) && compOpportunisticallyDependsOn(InstructionSet_AVX) ? YMM_REGSIZE_BYTES : XMM_REGSIZE_BYTES;
```
* The `size >= YMM_REGSIZE_BYTES` exists so we can still handle 16-31 byte copies via SIMD on hardware with 256-bit vector support, otherwise it looks like we'd fallback to standard `mov` copies here, which seems undesirable

`getSIMDVectorRegisterByteLength` is largely oriented around `System.Numerics.Vector<T>` which is variable sized and is only 256-bits if it can accelerate both integrals and floating-point (which requires AVX2). The JIT, in general, however supports `System.Runtime.Intrinsics.Vector256<T>` and so can work with 256-bit vectors if `AVX` is supported, even if `AVX2` is not.

This will also avoid any "issues" if we end up introducing support for `Vector512<T>` in the future (such as for AVX-512) and likewise allow `getSIMDVectorRegisterByteLength` to return `64`.

Reply:
I see, that makes sense. Ok, I'll update with your suggestion as well.